spark/src/test/java/org/apache/iceberg/actions/TestRewriteDataFilesAction.java

```diff
@@ -311,6 +325,128 @@ public void testRewriteLargeTableHasResiduals() {
     Assert.assertEquals("Rows must match", records, actualRecords);
   }
 
+  @Test
+  public void testBaseCombinedScanTaskKryoSerialization() throws Exception {
```

Comment:
Why add these tests to `TestRewriteDataFilesAction`? These aren't specific to that action and the code change is in core. I think it would be more better to add these tests to `TestDataFileSerialization`, since that is quite similar. That, or create a new test suite, `TestScanTaskSerialization`. Either one is fine with me.

Reply:
I think I missed to migrate comments from the origin PR, my bad.

> I decided to add the serde tests of BaseCombinedScanTask here, because it seems to require non-trivial efforts to create BasedCombinedScanTask by hand. I can move out if we'd like to have separate suite with manually crafted test objects.

So that was the plan to reduce lots of manual lines by hand, but now I realized the new test cases would just duplicate small code in TestRewriteDataFilesAction even it becomes a new test suite. I'll add a new test suite. Thanks!